apps/transport/lib/jobs/resource_history_jsonschema_validation_job.ex

```diff
@@ -16,36 +15,53 @@ defmodule Transport.Jobs.ResourceHistoryJSONSchemaValidationJob do
   """
   @impl Oban.Worker
   def perform(%Oban.Job{args: %{"resource_history_id" => resource_history_id}}) when is_integer(resource_history_id) do
+    ResourceHistorySchemaValidation.validate_resource_history_for_schema(resource_history_id, validator())
+  end
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{}) do
+    ResourceHistorySchemaValidation.enqueue_jobs_for_schema_type("jsonschema", __MODULE__, validator())
+  end
+
+  def validator, do: Transport.Validators.EXJSONSchema
+end
+
+defmodule Transport.Jobs.ResourceHistorySchemaValidation do
+  @moduledoc """
+  A module to share common methods used to identify `DB.ResourceHistory` with
+  a schema and validate them against a schema.
+
+  It deals with JSON Schema and Table Schema.
+  """
+  import Ecto.Query
+  alias Transport.Shared.Schemas.Wrapper, as: Schemas
+
+  def validate_resource_history_for_schema(resource_history_id, validator) do
     resource_history = DB.ResourceHistory |> DB.Repo.get!(resource_history_id)
 
-    if resource_history |> DB.MultiValidation.already_validated?(validator()) do
-      {:discard, "resource history #{resource_history_id} is already validated by #{validator_name()}"}
+    if resource_history |> DB.MultiValidation.already_validated?(validator) do
```

Comment:
Avec le système de mettre `%{"validation_performed" => false}` en cas de problème, la ressource va être considérée comme "already_validated" et ne sera pas revalidée. Si c'est parce que la payload de retour du validateur était indéchiffrable, c'est un comportement attendu. Mais si c'était dû à une indisponibilité temporaire du validateur, ça va nous embêter.

Une solution serait de dupliquer le code du validateur initial (Shared.Validation.TableSchemaValidator) plutôt que de l'appeler. Et du coup d'être libre de gérer plus finement ce qu'il se passe quand la validation échoue. Par exemple, pour une payload indéchiffrable, on met une ligne de multi_validation avec une erreur "impossible to decode json response", mais pour une impossibilité de joindre le serveur de validation (erreur 4XX ou 5XX), on n'insert pas de ligne dans la table de multi-validation.

En l'état si le serveur validata est down une journée, toutes les ressources qui auront tenté une validation vont se retrouver sans validation et ne pourront plus en avoir, sauf à forcer à la main leur revalidation lorsqu'on s'en rendra compte.

Qu'est ce que tu en penses ?

Reply:
Je te propose de gérer ceci dans une PR distincte, celle-ci me semble déjà assez étoffée.